### PR TITLE
Hubspot analytics

### PR DESCRIPTION
**Pull request by lachieh**
Some refactoring to add a little extra functionality in page categorization and move the logic into its own plugin folder.

### CODE REVIEW

File: packages/docusaurus-github-stars/src/utils/constants.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
+import { packageUpSync } from 'package-up';
 import fs from 'node:fs';
 
 function getPluginName() {
-  const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
+  const packageStr = packageUpSync({ cwd: __dirname });
+  if (!packageStr) throw new Error('Could not find package.json');
+  const packageJson = JSON.parse(fs.readFileSync(packageStr, 'utf-8')) as { name: string };
```
Comment (vados-cosmonic) [inline packages/docusaurus-github-stars/src/utils/constants.ts]: I could have sworn that Typescript had a thing for importing JSON files (https://www.typescriptlang.org/tsconfig/#resolveJsonModule ?), can we use that?

File: packages/plugin-hubspot-analytics/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@wasmcloud/plugin-hubspot-analytics",
+  "version": "0.0.0",
```
Comment (vados-cosmonic) [inline packages/plugin-hubspot-analytics/package.json]: A version might be nice here, even if it's `0.0.1`, though IMO `0.1.0` is also reasonable

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: packages/docusaurus-github-stars/src/utils/constants.ts (hunk 1 above)
Comment (lachieh) [reply packages/docusaurus-github-stars/src/utils/constants.ts]: Good call, a standard `require` call works. I also moved these utils into a reusable repo package.

File: packages/plugin-hubspot-analytics/package.json (hunk 2 above)
Comment (lachieh) [reply packages/plugin-hubspot-analytics/package.json]: If we decide to get them out into a registry we can set a version then but for now, this is the convention to indicate that it's a workspace-only package.

Comment (vados-cosmonic) [reply packages/plugin-hubspot-analytics/package.json]: TIL, thanks!

**Pull request merged**